cosmo_tester/test_suites/service_composition/shared_resource_test.py

```diff
@@ -20,3 +24,59 @@ def test_shared_resource(image_based_manager, ssh_key, logger, test_config):
         _verify_deployments_and_nodes(app, 2)
         _check_custom_execute_operation(app, logger)
         _verify_custom_execution_cancel_and_resume(app, logger)
+
+
+@pytest.fixture(scope='function')
+def managers(request, ssh_key, module_tmpdir, test_config, logger):
+    hosts = Hosts(ssh_key, module_tmpdir, test_config, logger, request, 2)
+    try:
+        hosts.create()
+        yield hosts.instances
+    finally:
+        hosts.destroy()
+
+
+def test_external_shared_resource_idd(managers, ssh_key, logger, test_config):
+    tenant = 'test_external_shared_resource'
+    local_mgr, external_mgr = managers
+
+    infra = _infra(external_mgr, ssh_key, logger, tenant, test_config)
+    app = _app(local_mgr, ssh_key, logger, tenant, test_config,
+               blueprint_name='shared_resource',
+               client_ip=external_mgr.private_ip_address)
+
+    logger.info('Deploying the shared resource on an external manager.')
+    infra.upload_blueprint()
+    infra.create_deployment()
+    logger.info('Deploying an application which uses the resource, '
+                'on the local manager.')
+    app.upload_and_verify_install()
+
+    logger.info('Verifying inter-deployment dependency on both managers')
+    with util.set_client_tenant(app.manager.client, tenant):
+        local_idd = app.manager.client.inter_deployment_dependencies.list()[0]
```

Comment:
Can't help but read this as local_if, probably have been to Cardiff (Caer Dydd) too often.